scripts/build_pkgdown.R

```diff
@@ -18,15 +19,81 @@ if (requireNamespace("PEcAn.logger", quietly = TRUE)) {
   }
 }
 
-logger("Building pkgdown docs for:", paste(packages, collapse = ", "))
+pkg_config <- function(pkg) {
+  pkgname <- desc::desc_get("Package", pkg)
+
+  list(
+    url = "https://pecanproject.github.io/",
+    home = list(
+      title = sprintf("%s Functions for PEcAn", pkgname),
+    ),
+    template = list(
+      bootstrap = 5,
+      bslib = list(
+        primary = "#0054AD", 
+        `border-radius` = "0.5rem",
+        `btn-border-radius` = "0.25rem"
+      ),
+      `light-switch` = TRUE, 
+    ),
+    navbar = list(
+      structure = list(
+        left = c("pecan_home", "reference", "news"),
+        right = c("search", "github", "light-switch") 
+      ),
+      components = list(
+        pecan_home = list(
+        text = "PEcAn Home",
+        href = "../../../index.html",
+        `aria-label` = "PEcAn Project Home"
+      ),
+        reference = list(
+        text = "Reference",
+        href = "reference/index.html"
+      ),
+        github = list(
+        icon = "fab fa-github",
+        href = "https://github.com/PecanProject/pecan",
+        `aria-label` = "GitHub"
+      )
+    )
+  ),
+    reference = list(
+      list(
+        title = "All Functions",
+        desc = "All functions exported by this package",
+        contents = list("matches('.*')")
+      )
+    ),
+    news = list(
+      text = "News",
+      href = "news/index.html"
+    ),
+    development = list(
+      mode = "auto"
+    )
+  )
+}
 
+logger("Building pkgdown docs for:", paste(packages, collapse = ", "))
 for (pkg in packages) {
   logger("Building pkgdown site for:", pkg)
   current_wd <- getwd()  
   tryCatch({
     if (!dir.exists(pkg)) {
       stop(paste("Package directory does not exist:", pkg))
     }
+    pkg_config_path <- file.path(pkg, "_pkgdown.yml")
+    pkg_config <- pkg_config(pkg)
+    # If _pkgdown.yml exists, merge with our config, otherwise create new
+    if (file.exists(pkg_config_path)) {
+      exist_config <- yaml::read_yaml(pkg_config_path)
+      # Merge configurations, preserving existing settings
+      merged_config <- modifyList(exist_config, pkg_config)
+      yaml::write_yaml(merged_config, pkg_config_path)
+    } else {
+      yaml::write_yaml(pkg_config, pkg_config_path)
+    }
```

Comment:
It's almost always a bad idea to allow hand-edits and autogenerated entries in the same file. Better to commit to the principle that _pkgdown.yml is generated by this script and regenerated as needed.

```suggestion
    yaml::write_yaml(pkg_config, pkg_config_path)
```